hibernate-core/src/main/java/org/hibernate/event/internal/DefaultEvictEventListener.java

```diff
@@ -111,7 +111,7 @@ protected void doEvict(
 
 		final PersistenceContext persistenceContext = session.getPersistenceContextInternal();
 		if ( persister.hasNaturalIdentifier() ) {
-			persistenceContext.getNaturalIdResolutions().handleEviction( object, key.getIdentifier(), persister );
+			persistenceContext.getNaturalIdResolutions().handleEviction( key.getIdentifier(), object, persister );
```

Comment:
This was the bug reported in issue https://hibernate.atlassian.net/browse/HHH-16010